File: src/haskell/util/Colors.hs
module Src.Haskell.Util.Colors(suc, norm, err, fine)where

black   = "\x11b[30m"
red     = "\x1b[31m"
green   = "\x1b[32m"
yellow  = "\x1b[33m"
blue    = "\x1b[34m"
magenta = "\x1b[35m"
cyan    = "\x1b[36m"
white   = "\x1b[37m"
reset   = "\x1b[0m"


suc cs   = green ++ cs ++ reset
norm cs  = blue ++ cs ++ reset
err cs   = red ++ cs ++ reset
fine cs  = yellow ++ cs ++ reset


main = do
  let hoge = fine "fuga"
  putStrLn hoge

{-
runghc src/haskell/src/Colors.hs
-}

File: src/rust/src/util/colors.rs
const BLACK:&'static str = "\x11b[30m"; 
const RED:&'static str = "\x1b[31m"; 
const GREEN:&'static str = "\x1b[32m"; 
const YELLOW:&'static str = "\x1b[33m";
const BLUE:&'static str = "\x1b[34m"; 
const MAGENTA:&'static str = "\x1b[35m"; 
const CYAN:&'static str = "\x1b[36m"; 
const WHITE:&'static str = "\x1b[37m"; 
const RESET:&'static str = "\x1b[0m";

pub fn suc(s: &str) -> String{ GREEN.to_string() + s + RESET }
pub fn fail(s: &str) -> String{ RED.to_string() + s + RESET }
pub fn norm(s: &str) -> String{ BLUE.to_string() + s + RESET }
pub fn fine(s: &str) -> String{ YELLOW.to_string() + s + RESET }